sentry/src/main/java/io/sentry/config/ClasspathPropertiesLoader.java

```diff
@@ -0,0 +1,46 @@
+package io.sentry.config;
+
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Loads {@link Properties} from the `sentry.properties` file located on the classpath. */
+final class ClasspathPropertiesLoader implements PropertiesLoader {
+  private final @NotNull String fileName;
+  private final @NotNull ClassLoader classLoader;
+  private final @NotNull ILogger logger;
+
+  public ClasspathPropertiesLoader(
+      @NotNull String fileName, @NotNull ClassLoader classLoader, @NotNull ILogger logger) {
+    this.fileName = fileName;
+    this.classLoader = classLoader;
+    this.logger = logger;
+  }
+
+  public ClasspathPropertiesLoader(@NotNull ILogger logger) {
+    this("sentry.properties", ClasspathPropertiesLoader.class.getClassLoader(), logger);
+  }
+
+  @Override
+  public @Nullable Properties load() {
+    try (InputStream inputStream = classLoader.getResourceAsStream(fileName)) {
+      if (inputStream != null) {
```

Comment:
do we need this `if` check? wouldn't throw on `getResourceAsStream` if an error?

Reply:
We do as it may return null if resource is not found https://docs.oracle.com/javase/7/docs/api/java/lang/ClassLoader.html#getResourceAsStream(java.lang.String)

Reply:
thanks